pkgs/games/atlauncher/default.nix

```diff
@@ -0,0 +1,33 @@
+{ fetchurl, jre, lib, makeWrapper, stdenv }:
+
+let
+  version = "3.4.28.1";
+in
+stdenv.mkDerivation rec {
+  pname = "atlauncher";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${version}/ATLauncher-${version}.jar";
+    sha256 = "sha256-IIwDMazxUMQ7nGQk/4VEZicgCmCR4oR8UYtO36pCEq4=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
```

Comment:
`runHook preInstall` at the beginning of installPhase and `runHook postInstall` at the end

it will help you avoid some sneaky bugs